src/lib.rs

```diff
@@ -366,10 +367,18 @@ impl Contract {
         new_community.set_default_admin();
         self.communities.insert(&new_community.handle, &new_community);
 
-        ext_devhub_community_factory::ext(get_devhub_community_factory())
+        let promise = ext_devhub_community_factory::ext(get_devhub_community_factory())
             .with_unused_gas_weight(1)
             .with_attached_deposit(CREATE_COMMUNITY_BALANCE)
-            .create_community_account(new_community.handle);
+            .create_community_account(new_community.handle.clone());
+
+        env::log_str("Community created successfully. Creating discussions account.");
+        promise.then(
+            ext_devhub_community::ext(get_devhub_discussions_factory(&new_community.handle))
+                .with_unused_gas_weight(2)
+                .with_attached_deposit(CREATE_DISCUSSION_BALANCE)
+                .create_discussions_account(),
+        );
```

Comment:
I would suggest the community contract to create the discussion subaccount on `new` instead of "micro-managing" the community implementation from the main contract.

Reply:
I'm trying to call `create_discussions_account` from the init function of the "announcements" contract (3). 
How can one attach a deposit? I thought an init function isn't payable.

Reply:
Does it have to be a method decorated with `#[init]` to do the initialization? Can't it just be a regular method that you call with a `#[payable]` decorator?

Reply:
@Tguntenaar Add `#[payable]` above or below the `#[init]` attribute and that is it: https://github.com/near/near-sdk-rs/blob/a12da50469c4a47640b6bd71ed4b433370978821/near-sdk-macros/src/core_impl/code_generator/item_impl_info.rs#L221-L231